x/attribute/keeper/keeper.go

```diff
@@ -217,13 +249,15 @@ func (k Keeper) UpdateAttribute(ctx sdk.Context, originalAttribute types.Attribu
 		if attr.Name == updateAttribute.Name && bytes.Equal(attr.Value, originalAttribute.Value) && attr.AttributeType == originalAttribute.AttributeType {
 			found = true
 			store.Delete(it.Key())
+			k.DecAddNameAddressLookup(ctx, attr)
```

Comment:
This is an odd flow ... an update with a decrement then an increment... While it might seem like there could be a break due to error handling in the marshal of the update ... this same error would prevent the transaction from committing at all so the delete would not happen.  Given this I don't see why we need to do a `-1/+1` in the same flow.

Reply:
The update allows for an address change which would mean a different key from the original.  So, I thought it would be appropriate just to decrement the address count when the old attribute is removed and increment it when the updated one is added.  I could do a check to see if the update would change what the lookup key would be and then do a dec of the old key and inc for the new key.

Reply:
Makes sense!